### ISSUE TITLE
Async component loading

### ISSUE THREAD

**Issue by kethan** [nanostores/nanostores#37]
Hi,

How can I lazy load components in the router using react or preact? And how can I do nested routing too?

Thanks

**Comment by ai** [comment]
> How can I lazy load components in the router using react or preact?

You can use the standard component lazy loading in React/Preact:

```js
const LazyPage = React.lazy(() => import('./LazyPage'))

if (route.path === '/lazy') {
  return <LazyPage />
}
```

> And how can I do nested routing too?

There is no simple API for it. Maybe we can add some advanced router for people who need this feature.

What is your use case?

**Comment by kethan** [comment]
ok, my use case is something like this.

```js
 <Route path="/" Component={AppPage}>
    <Route Component={MainPage} />
    <Route path="foo" Component={FooPage}>
      <Route path="bar" Component={BarPage} />
    </Route>
  </Route>
```

**Comment by ai** [comment]
Is it just for syntax sugar?

You can create a function which will convert tree structure routes to list or regexps:

```js
createRouter(treeRoutes({
  main: {
    foo: 'foo/:id'
  }
}))
```

**Comment by kethan** [comment]
> Is it just for syntax sugar?
> 
> You can create a function which will convert tree structure routes to list or regexps:
> 
> ```js
> createRouter(treeRoutes({
>   main: {
>     foo: 'foo/:id'
>   }
> }))
> ```

Yes, something similar. If possible these tree routes can be derived from nested components tags like above.

**Comment by kethan** [comment]
I am sorry this is closed. I don't know how to implement treeRoutes. May be I can use something like this returning multiple components for nesting
```js
<>
</>
<>
```

**Comment by ai** [comment]
I will try to add syntax sugar next week.

But it is just syntax sugar. You can start without it.

**Comment by kethan** [comment]
> I will try to add syntax sugar next week.
> 
> But it is just syntax sugar. You can start without it.

Yes, I got your point I am working on it thanks for guiding me.

**Comment by kethan** [comment]
If possible sometime similar like these nested routes I require for my next project. 
https://reach.tech/router/example/nested-routes

**Comment by ai** [comment]
React Routes uses React components to create a router. Nano Stores Router uses state manager to put logic.

So you can’t use examples from React component routers for API. The basement is very different.
